tests/operators/test_great_expectations.py

```diff
@@ -0,0 +1,149 @@
+"""
+Unit test module to test Great Expectations Operators.
+
+Requirements can be installed in, for instance, a virtual environment, with::
+
+    pip install -r requirements`
+
+and tests can be run with::
+
+    pytest -p no:warnings
+"""
+
+from contextlib import contextmanager
+import logging
+import os
+from pathlib import Path
+import unittest
+import unittest.mock as mock
+
+from airflow.exceptions import AirflowException
+from great_expectations_provider.operators.great_expectations import (
+    GreatExpectationsOperator,
+)
+from great_expectations.exceptions.exceptions import GreatExpectationsError
+
+
+log = logging.getLogger(__name__)
+
+# Set relative paths for Great Expectations directory and sample data
+base_path = Path(__file__).parents[2]
+good_data_file = os.path.join(
+    base_path, "include", "data/yellow_tripdata_sample_2019-01.csv"
+)
+bad_data_file = os.path.join(
+    base_path, "include", "data/yellow_tripdata_sample_2019-02.csv"
+)
+ge_root_dir = os.path.join(base_path, "include", "great_expectations")
+
+
+class TestGreatExpectationsOperator(unittest.TestCase):
+    def test_great_expectations_operator__valid_expectation_suite(self):
+        operator = GreatExpectationsOperator(
+            task_id="task_id",
+            expectation_suite_name="taxi.demo",
+            batch_kwargs={"path": good_data_file, "datasource": "data__dir"},
+            data_context_root_dir=ge_root_dir,
+        )
+        result = operator.execute(context={})
+        log.info(result)
+        self.assertTrue(result["success"])
+
+    def test_great_expectations_operator__invalid_expectation_suite(self):
+        operator = GreatExpectationsOperator(
+            task_id="task_id",
+            expectation_suite_name="invalid-expectation-suite.name",
+            batch_kwargs={"path": good_data_file, "datasource": "data__dir"},
+            data_context_root_dir=ge_root_dir,
+        )
+        self.assertRaises(GreatExpectationsError, operator.execute, context={})
+
+    def test_great_expectations_operator__valid_checkpoint(self):
+        # This is ugly, but the older version of checkpoint configs that this supports
+        # doesn't seem to allow relative paths for the data assets being validated.
```

Comment:
If you look at the main `great_expectations` repo, you'll see similar, but maybe more realized patterns that might be helpful, for instance in `tests/data_context/test_data_context.py::test_render_full_static_site_from_empty_project` and in `tests/integration/test_script_runner.py::_execute_integration_test`

Reply:
I'd, personally, prefer to use `pytest` along with all the features and fixtures it provides (like `tmp_path`), but there was a desire by Plinio to support unit tests being run with a simple `python -m unittest ...` and to not directly rely on `pytest`.

If you'd prefer that we just assume `pytest` is being used to drive the tests, great.  Let me know and I can migrate the unit tests to `pytest` style tests.

Reply:
That's great context to have. I hear Plinio's point, but we do have the eventual goal of incorporating this project within the main the Great Expectations codebase (though we do not have a definite timeline for this). At that point, we'll need to migrate to the standard `pytest` way. My inclination is to just do that now since it makes things easier for this type of maintenance in the short-term and the long-term, unless it makes things significantly more difficult for the Astronomer folks (and then it would be good to discuss).

Reply:
OK, so should we get Plinio's feedback now or just make the changes first?  I'm fine either way...

Reply:
I think let's check with @pgzmnk to confirm first.

Reply:
While we _encourage_ `unittest` as a standard across Airflow providers, using `pytest` makes sense given the reasoning above.

Reply:
Thanks, @pgzmnk.  I went ahead and moved to the `pytest` only tests.

CC @talagluck

Reply:
Thanks, both!